certbot/docs/contributing.rst

```diff
@@ -338,17 +338,57 @@ it was not installed properly.
 Once you've finished your plugin and published it, you can have your
 users install it system-wide with `pip install`. Note that this will
 only work for users who have Certbot installed from OS packages or via
-pip. Users who run `certbot-auto` are currently unable to use third-party
-plugins. It's technically possible to install third-party plugins into
-the virtualenv used by `certbot-auto`, but they will be wiped away when
-`certbot-auto` upgrades. If you'd like your plugin to be used alongside
-the Certbot snap, you will also have to publish your plugin as a snap.
-Certbot's DNS plugins and the README file in ``tools/snap/`` provide a
-starting reference for how to do this.
+pip.
 
 .. _`setuptools entry points`:
     https://setuptools.readthedocs.io/en/latest/pkg_resources.html#entry-points
 
+Writing your own plugin snap
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you'd like your plugin to be used alongside the Certbot snap, you
+will also have to publish your plugin as a snap. Plugin snaps are
+regular confined snaps, but normally do not provide any "apps"
+themselves. Plugin snaps export loadable Python modules to the Certbot
+snap.
+
+When the Certbot snap runs, it will use its version of Python and prefer
+Python modules contained in its own snap over modules contained in
+external snaps. This means that your snap doesn't have to contain things
+like an extra copy of Python, Certbot, or their dependencies, but also
+that if you need a different version of a dependency than is already
+installed in the Certbot snap, the Certbot snap will have to be updated.
+
+Certbot plugin snaps expose their Python modules to the Certbot snap via
+a `snap content interface`_ named ``certbot-1``. The Certbot snap also
```

Comment:
Hm, I'd say that the interfaces are "named" `plugin` and `certbot-metadata`; `certbot-1` and `metadata-1` are more version strings. You can tell because the name you use on the command line to `connect` them is `:plugin`.

Reply:
Good catch. https://snapcraft.io/docs/content-interface even refers to the values you mention as the plug/slot name. I made this more explicit by changing it to:

> ...where `certbot-1` is the value for the `content` attribute.

and 

> using the `content` identifier `metadata-1`.

I wanted to mention these values because they must be the same for it to work.